tests/src/Appender/RollingFileAppenderTest.cs

```diff
@@ -240,13 +252,52 @@ public void RollingCombinedWithPreserveExtension()
 			VerifyFileCount(2, true);
 		}
 
+		[Test]
+		public void RollingBySizePreserveExtensionAndWithFolder()
+		{
+			_root = ((Repository.Hierarchy.Hierarchy)Utils.GetRepository()).Root;
+			_root.Level = Level.All;
+			PatternLayout patternLayout = new PatternLayout();
+			patternLayout.ActivateOptions();
+
+			RollingFileAppender roller = new RollingFileAppender();
+			roller.StaticLogFileName = false;
+			roller.Layout = patternLayout;
+			roller.AppendToFile = true;
+			roller.RollingStyle = RollingFileAppender.RollingMode.Size;
+			roller.MaxSizeRollBackups = 2;
+			roller.CountDirection = 1;
+			roller.PreserveLogFileNameExtension = true;
+			roller.MaximumFileSize = "5KB";
+			roller.File = c_fileNameWithFolder; 
+			roller.ActivateOptions();
+			_root.AddAppender(roller);
+
+			_root.Repository.Configured = true;
+
+			for (int i = 0; i < 1000; i++)
+			{
+				StringBuilder s = new StringBuilder();
+				for (int j = 50; j < 100; j++)
+				{
+					if (j > 50)
+					{
+						s.Append(" ");
+					}
+					s.Append(j);
+				}
+				_root.Log(Level.Debug, s.ToString(), null);
+			}
+			VerifyFileCount(c_fileNameWithFolder, roller.MaxSizeRollBackups + 1, true);
```

Comment:
This should probably read as `VerifyFileCount(roller.MaxSizeRollBackups + 1, true);`